### ISSUE TITLE
Determine Element reach

### ISSUE THREAD

**Issue by denvasden** [fisshy/react-scroll#283]
Hi!

Are there any ability to determine that the element has been reached and apply some class to this element using react-scroll?

Thanks a lot!

It's not the firs time I post this question.
The first time I've recieved and answer:

> Yes, theres a begin/end event.

var Scroll  = require('react-scroll');
var Events     = Scroll.Events;

Events.scrollEvent.register('begin', function(to, element) {
      console.log("begin", arguments);
});

Events.scrollEvent.register('end', function(to, element) {
      console.log("end", arguments);
});

> 

But where do I need to register this events.

On an element or on an parent element.
I've tried to use both of this approaches and nothing
happend.

Callbacks don't fire during scroll.

**Comment by fisshy** [comment]
Hey!

If you look at this example, you can see that it's being used in the componentDidMount of the parent element.

**Comment by denvasden** [comment]
@fisshy yes, I've seen this.

Looks like this events emits only when you call scrollTo method.

Seems like there is no way to determine that the element has been reached and apply class to this element.

I mean I have an element in the bottom of my page and I want to apply some class to this element when I reach it during the scroll.

I can see only Link <-> Element relation.

**Comment by fisshy** [comment]
Yes you are right, one solution could be to implement an "onActive" event on the Links. 

Let me look into it and come up with a solution for you :)

**Comment by fisshy** [comment]
I believe you should be able to use this

```
<Link 
      onSetActive={this.handleSetActive}
      onSetInactive={this.handleSetInactive}
>
  Your name
</Link>
```
